### PR TITLE
Flish/#1225 default mime type for attachments

### PR DESCRIPTION
**Pull request by flish**
### Description of Change ###

- Fixed Mimetypes being null and throwing a null exception if mimetype not found on iOS / TVOS
- default Mimetype if not found (null) to be "application/octet-stream" as suggested

No new tests or documentation changes required, sample / test app can be provided.  Simple test is to try an attach a file with .log or .db extension to an email, which are not dound as mimetypes so woudl not be attached (exception)

```
var message = new EmailMessage
{
	Subject = "TEST", Body = "BODY", To = new List<string> {"help@xyroh.com"}
};

message.BodyFormat = EmailBodyFormat.PlainText;
if (File.Exists(attachmentPath))
{
	Console.WriteLine("FILE " + attachmentPath + " FOUND");
	message.Attachments.Add(
		new EmailAttachment(attachmentPath)); //works as long as extension is txt (mime types!)
}

try
{
	foreach (var attach in message.Attachments)
	{
		Console.WriteLine("Type " + attach.ContentType.ToString());
	}
	await Email.ComposeAsync(message);
}
```

### Bugs Fixed ###

- Related to issue #1225 

Provide links to issues here. Ensure that a GitHub issue was created for your feature or bug fix before sending PR.

### API Changes ###

List all API changes here (or just put None), example:

None

### Behavioral Changes ###

If you were checking for null as as the ContentType (on non iOS platforms) then you will now always get a result, iOS / tvOS was throwing an exception anyway

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Xamarin.Essentials/FileSystem/FileSystem.ios.tvos.watchos.cs [hunk 1]
```diff
@@ -50,7 +50,14 @@ internal static string PlatformGetContentType(string extension)
 
             var id = UTType.CreatePreferredIdentifier(UTType.TagClassFilenameExtension, extension, null);
             var mimeTypes = UTType.CopyAllTags(id, UTType.TagClassMIMEType);
-            return mimeTypes.Length > 0 ? mimeTypes[0] : null;
+            if (mimeTypes != null && mimeTypes.Length > 0)
```
Comment (jamesmontemagno) [inline Xamarin.Essentials/FileSystem/FileSystem.ios.tvos.watchos.cs:53]: return (mimesTypes?.Length ?? 0) > 0 ? mimeTypes[0] : null;

File: Xamarin.Essentials/FileSystem/FileSystem.shared.cs [hunk 2]
```diff
@@ -73,7 +73,9 @@ internal string GetContentType()
 
             // we haven't been able to determine this
             // leave it up to the sender
-            return null;
+            // return null;
+            // Instead of returning null return a catch all mime-type - see #1225
```
Comment (jamesmontemagno) [inline Xamarin.Essentials/FileSystem/FileSystem.shared.cs:77]: don't like to put in issue numbers in the code, as we have blame to help us out and link back.

Do we want it to be octet-stream? does everythign support that?

Comment (flish) [reply Xamarin.Essentials/FileSystem/FileSystem.shared.cs:77]: Fair enough ref comments,  octet-stream was suggested in the comments and is as generic as it comes, there are no real application handlers for it so gets attached / downloaded and up to the end user to handle

File: Xamarin.Essentials/FileSystem/FileSystem.ios.tvos.watchos.cs (hunk 1 above)
Comment (flish) [reply Xamarin.Essentials/FileSystem/FileSystem.ios.tvos.watchos.cs:53]: I’m old school readable code, like my whitespace :-] Will tweak in the am if no more suggestions

File: Xamarin.Essentials/FileSystem/FileSystem.shared.cs (hunk 2 above)
Comment (jamesmontemagno) [reply Xamarin.Essentials/FileSystem/FileSystem.shared.cs:77]: I think if we just test it out on iOS/Android that then would be good... just want to does work... we can just document it and recommend people set it manually.

File: Xamarin.Essentials/FileSystem/FileSystem.ios.tvos.watchos.cs (hunk 1 above)
Comment (mattleibow) [reply Xamarin.Essentials/FileSystem/FileSystem.ios.tvos.watchos.cs:53]: You can also just do

return mimesTypes?.Length > 0 ? mimeTypes[0] : null;

Null never matches int, so will always be false.

**Comment by flish** [comment]
@jamesmontemagno  @mattleibow - just pushed the changes, there's a Xam forms sample project at https://xyroh-my.sharepoint.com/:u:/p/af/ETfQysgpf15KksFR8uRVeWQBZJPioTui9Ijv2E-bkA1Ifw?e=jCEe1B if you wnat, but can confirm the octet-stream mime type works fine as a default on iOS and Android, just up to the individual device to do something specific with it

**Pull request merged**